TicTacToe/TicTacToe.cs

```diff
@@ -14,63 +14,102 @@ class Program
 
         public static void Main()
         {
+            int tieInt = 0;     //if reaches 9, game ends
             do
             {
+                int[] newArr = new int[2];
+                DrawBoard();
+                newArr = GetInput();
+                if(PlaceMark(newArr[0],newArr[1]) == true && playerTurn == "X")
+                {
+                    board[newArr[0]][newArr[1]] = playerTurn;
+                    playerTurn = "O";
+                    tieInt++;
+                }
+                else if(PlaceMark(newArr[0],newArr[1]) == true && playerTurn == "O")
+                {
+                    board[newArr[0]][newArr[1]] = playerTurn;
+                    playerTurn = "X";
+                    tieInt++;
+                }
+                else
+                {
+                    newArr = GetInput();        //when user input is already taken or out of bound
+                }
                 DrawBoard();
-                GetInput();
 
-            } while (!CheckForWin() && !CheckForTie());
+            } while (!CheckForWin() && tieInt < 9); //game ending condition
 
             // leave this command at the end so your program does not close automatically
             Console.ReadLine();
         }
-
-        public static void GetInput()
+        
+        public static int[] GetInput()           //return two ints that store user inputs
         {
             Console.WriteLine("Player " + playerTurn);
             Console.WriteLine("Enter Row:");
             int row = int.Parse(Console.ReadLine());
             Console.WriteLine("Enter Column:");
             int column = int.Parse(Console.ReadLine());
+            int[] taken = new[] {row, column};
+            return taken;
         }
-
-        public static void PlaceMark(int row, int column)
+        public static bool PlaceMark(int row, int column) //return true if possible to put a mark
         {
-        // your code goes here
+            // your code goes here
+            if(row > 2 || column > 2)
+                return false;
+            if(board[row][column] == "X" || board[row][column] == "O")
+                return false;
+            return true;
         }
 
         public static bool CheckForWin()
         {
             // your code goes here
-
+            if (HorizontalWin() || VerticalWin() || DiagonalWin())
+                return true;
             return false;
         }
 
-        public static bool CheckForTie()
-        {
-            // your code goes here
-
-            return false;
-        }
+        // public static bool CheckForTie() //have my own tie condition
+        // {
+        //     // your code goes here
+        //     if (HorizontalWin() == false || VerticalWin() == false || DiagonalWin() == false)
+        //         return true;
+        //     return false;
+        // }
         
         public static bool HorizontalWin()
```

Comment:
The methods that check for win conditions use playerTurn when checking, but by that point the player's turn has switched to the other player, so it will capture a win 1 turn too late.